SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -464,7 +464,11 @@ void USpatialWorkerConnection::OnConnectionFailure()
 			const FString ErrorMessage(UTF8_TO_TCHAR(OpList->ops[i].disconnect.reason));
 			AsyncTask(ENamedThreads::GameThread, [this, ErrorMessage]
 			{
-				GetSpatialNetDriverChecked()->HandleOnConnectionFailed(ErrorMessage);
+				UGameInstance* GameInstance = Cast<UGameInstance>(GetOuter());
+				if (GEngine != nullptr && GameInstance->GetWorld() != nullptr)
+				{
+					GEngine->BroadcastNetworkFailure(GameInstance->GetWorld(), GetSpatialNetDriverChecked(), ENetworkFailure::PendingConnectionFailure, *ErrorMessage);
```

Comment:
Arguably not part of this PR, but we no longer need to get the op list to find the disconnect message. `Worker_Connection_GetConnectionStatusDetailString` is the correct way to deal with that.
Eventually this might matter as some connection failures are retry-able and some are not. It could also allow us to not expose the connection, or anything representing a worker until we know that the connection has succeeded (although right now I don't think that buys us anything).